backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java

```diff
@@ -76,6 +76,13 @@ public CodeVersionHandler(Executor executor, FutureJdbi jdbi) {
             maybeSnapshotId -> {
               if (maybeSnapshotId.isPresent()) {
                 if (request.getOverwrite()) {
+                  jdbi.useHandle(
+                      handle ->
+                          handle
+                              .createUpdate(
+                                  "UPDATE experiment_run SET code_version_snapshot_id = null WHERE id=:run_id")
+                              .bind("run_id", request.getId())
+                              .execute());
```

Comment:
we can not delete code version snapshot directly because experimentRun have foreign key reference so first need to remove code_version_snapshot_id from experimentRun and then we can delete snapshot. so fixed it here.